src/datachain/lib/file.py

```diff
@@ -315,6 +316,35 @@ def get_fs(self):
         """Returns `fsspec` filesystem for the file."""
         return self._catalog.get_client(self.source).fs
 
+    def resolve(self) -> "File":
+        """
+        Resolve a File object by checking its existence and updating its metadata.
+
+        Returns:
+            File: The resolved File object with updated metadata.
+        """
+        if self._catalog is None:
+            raise RuntimeError("Cannot resolve file: catalog is not set")
+
+        client = self._catalog.get_client(self.source)
+        try:
+            info = client.info(self.path)
+
+            self.size = info.get("size", 0)
+            self.etag = info.get("etag", "")
+            self.version = info.get("version", "")
+            self.is_latest = info.get("is_latest", True)
+            self.last_modified = info.get("last_modified", datetime.now(timezone.utc))
+        except Exception as e:  # noqa: BLE001
+            self.size = -1
```

Comment:
Please use defaults from the File field definitions like size=0, is_latest=True.